include/eve/traits/overload/default_behaviors.hpp

```diff
@@ -0,0 +1,107 @@
+//======================================================================================================================
+/*
+  EVE - Expressive Vector Engine
+  Copyright : EVE Project Contributors
+  SPDX-License-Identifier: BSL-1.0
+*/
+//======================================================================================================================
+#pragma once
+
+#include <eve/concept/value.hpp>
+#include <eve/detail/skeleton.hpp>
+#include <eve/detail/function/conditional.hpp>
+
+//======================================================================================================================
+// Overload's helpers
+//======================================================================================================================
+namespace eve::detail
+{
+  //====================================================================================================================
+  // Internal concept checking if a default behavior is defined for a given callable
+  //====================================================================================================================
+  template<typename T, typename Arch, typename... Args>
+  concept supports_default_behavior = requires(Arch const& arch, Args&&... args)
+  {
+    { T::behavior::call(arch,EVE_FWD(args)...) };
+  };
+
+  //====================================================================================================================
+  // Aggregates and handles the default behavior of a given callable when no overload are found
+  // Reports hard error if no overload nor default behavior are available for T.
+  //====================================================================================================================
+  template<typename T, typename...>
+  struct default_behavior
+  {
+    template<typename Arch, typename... Args>
+    EVE_FORCEINLINE
+    static constexpr auto call(Arch const& arch, Args&&... args) requires(supports_default_behavior<T,Arch,Args...>)
+    {
+      return T::behavior::call(arch,EVE_FWD(args)...);
+    }
+
+    template<typename Arch, typename... Args>
+    EVE_FORCEINLINE
+    static constexpr auto call(Arch const&, Args&&...) requires(!supports_default_behavior<T,Arch,Args...>)
+    {
+      static_assert(supports_default_behavior<T, Arch, Args...>, "[EVE] - Incomplete callable object implementation");
```

Comment:
I don't understand this error message I think

Reply:
If you ends up there, it means you have given a callable a defautl behavior but you didnt implemented the correct function in it. So instead of ending up with  a brutal hard error, we notify the user its default behavior is incomplete.

I am open to get a better message

Reply:
this class is an intermediate steps that gather all possible defautl behavior (like being a constant, or being an elementwise operation) and provide an implementation so we don't have to overload soem familly of functions 1000x times doing the same thing. See the constant example.

Reply:
I don't see why you need this - look at my overload comments